pandas/_testing/asserters.py

```diff
@@ -455,7 +455,7 @@ def repr_class(x):
     raise_assert_detail(obj, msg, repr_class(left), repr_class(right))
 
 
-def assert_attr_equal(attr: str, left, right, obj: str = "Attributes"):
+def assert_attr_equal(attr: str, left, right, obj: str = "Attributes") -> bool:
```

Comment:
We had an issue about this previously (can't find it immediately), but all these `assert_*` functions should `-> None`. Since this function isn't publicly documented, happy to just have this refactored to not return bools